pkg/test/resource_creator.go

```diff
@@ -52,7 +54,28 @@ func (ctx *TestCtx) GetNamespace() (string, error) {
 	return ctx.namespace, nil
 }
 
-func (ctx *TestCtx) createFromYAML(yamlFile []byte, skipIfExists bool, cleanupOptions *CleanupOptions) error {
+// SetNamespace sets a static namespace for the current context. If the specifed namespace does not exist,
+// it will be created and a cleanup function for the new namespace will be added to the context.
+func (ctx *TestCtx) SetNamespace(namespace string) error {
+	ctx.namespace = namespace
+	namespaceObj := &core.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ctx.namespace}}
+	_, err := Global.KubeClient.CoreV1().Namespaces().Create(namespaceObj)
+	// do not add a cleanup function if the namespace already exists
+	if apierrors.IsAlreadyExists(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	ctx.AddCleanupFn(func() error {
+		return Global.KubeClient.CoreV1().Namespaces().Delete(ctx.namespace, metav1.NewDeleteOptions(0))
+	})
+	return nil
+}
+
+// CreateFromYAML takes a raw yaml file and creates the resource(s) in it in the cluster and adds cleanup functions
+// for the created resource(s).
+func (ctx *TestCtx) CreateFromYAML(yamlFile []byte, skipIfExists bool, cleanupOptions *CleanupOptions) error {
```

Comment:
If we have the `InitializeClusterResources` do we really need to expose this function?

Reply:
This is useful when users want to create a resource defined in a YAML file that is not set as a NamespacedResource (a CR for example). They could technically work around that like we originally did in the memcached test, but it's not clean or obvious how to do: https://github.com/operator-framework/operator-sdk/pull/1205/files#diff-62b541caac15c49bbed3723129bf478eL385